### DISCUSSION TITLE
Clearing Authstore in Next13

### DISCUSSION THREAD

**Discussion by LoganPikeDev** [pocketbase/pocketbase#1857] [Q&A]
I am trying to use an API route in next13 to log users out of the authstore, however, the issue I am running into is that I cannot access the cookies to set the authstore and clear it. I have tried the documented cookies method from Next/Headers but it keeps saying that the pages directory is not a server component. Does anyone have an easier way of reading cookies in API routes. Thank you

**Comment by arslanastral** [comment]
This doesn't seem to be a pocketbase issue. Are you using SSR and following the [SSR guide for Nextjs](https://github.com/pocketbase/js-sdk#ssr-integration)? Next.js 13 cookies function in a server component is **read-only** and server components only work inside [app](https://beta.nextjs.org/docs/routing/fundamentals#the-app-directory) directory. The [docs ](https://beta.nextjs.org/docs/api-reference/cookies) mention this:

> The cookies function allows you to read the HTTP incoming request cookies from a [server component](https://beta.nextjs.org/docs/rendering/server-and-client-components). It is read-only, meaning you cannot set / delete the outgoing request cookies.

>The Next.js team at Vercel is working on adding the ability to set cookies in addition to the cookies function.

**Comment by jaywei1234** [comment]
I have the same issue with you. I am using the exportToCookie but it cant store the auth to the cookie after logging in. And I can't find a solution for this.
